docs/dev/compiler-toolchain/vyper.md

```diff
@@ -0,0 +1,74 @@
+# Vyper
+
+The compiler we provide as a part of our toolchain is called [zkvyper](https://github.com/matter-labs/zkvyper-bin). It
+operates on LLL IR and metadata received from the underlying [vyper](https://docs.vyperlang.org/en/latest/index.html) compiler,
+which must be available in `$PATH`, or the path must be explicitly passed via the CLI (command-line interface).
```

Comment:
1. No-no, it's LLL, not LLVM.

2. I wasn't sure about `on our toolchain`, as I'm not really good at prepositions 😅 So I asked ChatGPT:
```
The Vyper compiler on our toolchain is called zkvyper.
What does the preposition `on` mean here? How to change the preposition so it will tell that the Vyper compiler is actually a part of our toolchain?
```
```
In this context, the preposition "on" is being used to indicate that the Vyper compiler is a component or tool that is installed and available within the toolchain, much like an add-on or plugin.

If you want to indicate that the Vyper compiler is actually a part of your toolchain and not just an installed component, you could use a different preposition that conveys this relationship more explicitly. For example, you could say:

"The Vyper compiler, which is an integral part of our toolchain, is called zkvyper."

This phrasing emphasizes that the Vyper compiler is not just an add-on or component of the toolchain, but rather an essential part of it.
```
Since it is not an installed component, but a required part, we should keep the original if you agree with ChatGPT 😸

Reply:
updated

Reply:
Just an IR Vyper uses alike to Yul and EVMLA in Solidity.